### ISSUE TITLE
Urho3D build failed ( PhysicsWorld.h )

### ISSUE THREAD

**Issue by Sheridan** [urho3d/Urho3D#2399]
I am try make gentoo ebuild for Urho3D (HEAD version of repo) with using system libraryes (i am strip third party libraryes from code) , but build failed on
https://github.com/urho3d/Urho3D/blob/9c7ea246ca6132053d07bd30429595d9a59d15fd/Source/Urho3D/Physics/PhysicsWorld.h#L144

with error:

```
...Source/Urho3D/Engine/../Physics/PhysicsWorld.h:144:10: error: ‘bool Urho3D::PhysicsWorld::isVisible(const btVector3&, const btVector3&)’ marked ‘override’, but does not override
     bool isVisible(const btVector3& aabbMin, const btVector3& aabbMax) override;
          ^~~~~~~~~
```
I understand correctly that this is due to different versions of Bullet?

**Comment by weitjong** [comment]
Not just that, a number of third-party libs that we embed in our source repo are actually our “fork”s that contain our own local changes. We do not support building Urho3D in the manner you described.

**Comment by iSLC** [comment]
Try removing the override keyword. But I'm sure you'll be facing the same issue on the next occurrence. And there are quite a few of them.

**Comment by Sheridan** [comment]
Next problem... When install, Urho try overvrite files from another package
```
 * Detected file collision(s):
 * 
 *      /usr/bin/isql
 *      /usr/lib/debug/usr/bin/isql.debug
 * 
 * Searching all installed packages for file collisions...
 * 
 * Press Ctrl-C to Stop
 * 
 * dev-db/unixODBC-2.3.5-r1:0::gentoo
 *      /usr/bin/isql
 *      /usr/lib/debug/usr/bin/isql.debug
 * 
```

**Comment by weitjong** [comment]
Are you saying our “isql” tool would **still** be installed to “/usr/bin”, even though we have by default configured all our build artifacts to **supposedly** be installed to “/usr/local/bin”? We do that exactly to avoid clashing with other OS level system packages. If you have modified our build script to change its install destination to “/usr” instead of “/usr/local” then you have to take care.

**Comment by weitjong** [comment]
And also I think usually all our tools are installed in “tool” directory under our own “bin” dir, so I really don’t understand why there is a collision. Unless you have not stated that you have modified our build script for your own need.

**Comment by Sheridan** [comment]
I did not make any changes in the paths, it can be seen in the [my urho3d-9999.ebuild](https://gist.github.com/Sheridan/90c0426d7f91395c76bc0bd7a1b1aa26). 
I rarely write ebuilds, but maybe gentoo itself substitutes in cmake the parameters that set the package in /usr

In fact, installing a package by default in a /usr/local block is a bit of a violation of "linux way" rules.
* /usr - for packages installed by the package manager.
* /usr/local - for programs installed manually, without a package manager (make, make install ...)
* /opt/package_folder - for programs or packages that do not comply with unix way and all bring with them. For example, often there are programs that come to Linux from Windows. For example Skype.

Naturally, there are no complaints to you, since I can configure the ebuild so that it installs the Urho3D in /opt/Urho3D, but on the other hand, I wanted to show you that you have the package installed a little wrong. At a minimum, other packages can be installed on the client computer in /usr/local (compiled manually), which binaries during installation can rewrite (without notifying the user what the binary will replace)

Sorry for my English...

**Comment by Sheridan** [comment]
Please note that when a program is pacckaged in, the package automatically becomes "system", as it is managed by the package manager. And therefore it should be installed (in our case) in /usr

**Comment by weitjong** [comment]
This is not the first time I have been pointed out about the mistake. The offending line in question is here.
https://github.com/urho3d/Urho3D/blob/master/CMakeLists.txt#L143-L145

In the past when we released Urho3D, we also provided the Debian *.deb and RedHat *.rpm; but I didn't want those to be considered as system packages and therefore (perhaps wrongly) decided to alter the destination path to "/usr/local". However, I purposely left a note there to let system package maintainer, like yourself, to patch the script as necessary. But like I said earlier, once you modify the script, you are on your own because we don't have the man power to support everyone or to keep everyone happy.

I recently have to modify our CI build so that it does not produce the Debian or RedHat packages anymore for other non-related reason. It means the future Urho3D releases won't have Debian and RedHat packages provided as well. And it also means there is no more reason for me to keep that offending line. I will make a commit to remove it at the soonest.

BTW, I have not seen an ebuilds script before, so I could not comment anything on that. But I really admire Gentoo users building everything from source yourself. Sorry for my English too, it is also not my first or second language. I have to write very long to get my message across (hopefully) :P

**Comment by Sheridan** [comment]
>> ...once you modify the script...

No, currently cmake scripts (and all other content) unmodified. For example, ebuild with sources modifications look like this: [urho3d-1.6.ebuild](https://github.com/orsonteodoro/oiledmachine-overlay/blob/master/games-engines/urho3d/urho3d-1.6.ebuild#L146)

**Comment by weitjong** [comment]
I have pushed a commit to remove the offending line. Both the "CMAKE_INSTALL_PREFIX" and "CPACK_PACKAGING_INSTALL_PREFIX" variables are left as they are defaulted by CMake. Since I don't think ebuild uses `cpack`, you just need to ensure the "CMAKE_INSTALL_PREFIX" variable is set to "/usr" before invoking `cmake` in your script. However, most likely the emerge/ebuild build system has done that for you. So, I believe that part is good.

> And also I think usually all our tools are installed in “tool” directory under our own “bin” dir, so I really don’t understand why there is a collision. Unless you have not stated that you have modified our build script for your own need.

I have time to double check it now. I was wrong in earlier comment. The "isql" executable is compiled using target compiler toolchain, so it is not a "host tool" which is compiled using host compiler toolchain. In our build system we have configured all the host tools to be resided in "bin/tool", while all the standalone target executables (like "isql" from SQLite library or "luajit" from LuaJIT library) to be resided in the "bin" directory, like the rest of target executables. If that is not desirable then I am afraid you or we have to patch our CMake script. The relevant lines are here.

https://github.com/urho3d/Urho3D/blob/master/Source/ThirdParty/SQLite/CMakeLists.txt#L50-L70

Perhaps, rename the executable name to something else. For this one, I am fine to support you to make the necessary change.
